### PR TITLE
[TOAZ-184] Integrate Landing Zone database

### PR DESCRIPTION
**Pull request by zykonda**
TOAZ-172 implementation.
Unit tests pending.

### CODE REVIEW

File: scripts/connect-sql.sh [hunk 1]
```diff
@@ -31,6 +31,11 @@ case $1 in
     dbuser=$(<"${configdir}/stairway-db-username.txt")
     dbpw=$(<"${configdir}/stairway-db-password.txt")
     ;;
+  landingzone)
+    db=$(<"${configdir}/landingzone-db-name.txt")
+    dbuser=$(<"${configdir}/landingzone-db-username.txt")
+    dbpw=$(<"${configdir}/landingzone-db-password.txt")
+    ;;
   *)
     echo "Specify wsm or stairway to choose which database to connect to"
```
Comment (sergiygetlin) [inline scripts/connect-sql.sh:40]: should we add landingzone here as well? As you did below?

File: service/src/main/java/bio/terra/landingzone/db/LandingZoneDao.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+/** LandingZoneDao includes operations on the landing zone tables. */
+@Component
+public class LandingZoneDao {
+  /** SQL query for reading landing zone records. */
+  private static final String LANDINGZONE_SELECT_SQL =
+      "SELECT landingzone_id, spend_profile, template_id, template_version, properties"
```
Comment (sergiygetlin) [inline service/src/main/java/bio/terra/landingzone/db/LandingZoneDao.java:27]: Maybe definition is better than template? You proposed it initially. And for landing zone listing rest api we are using path you've suggested - /landingzones/definitions.. So, I am thinking maybe it's better to use definition here as well..

File: service/src/main/resources/landingzone.db/changesets/20220722_revised_initial_schema.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# The schema for a landing zone
```
Comment (sergiygetlin) [inline service/src/main/resources/landingzone.db/changesets/20220722_revised_initial_schema.yaml:1]: Do we need tenantId, subscriptionId here?

Comment (sergiygetlin) [reply service/src/main/resources/landingzone.db/changesets/20220722_revised_initial_schema.yaml:1]: Disregard it please

File: scripts/clean_landingzone.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+-- Remove all of the data from all tables; leave the schema intact.
+TRUNCATE landingzone CASCADE;
```
Comment (ddietterich) [inline scripts/clean_landingzone.sql:2]: Is this a development convenience or is it used somewhere in the code?

**Review by ddietterich — CHANGES_REQUESTED** [review]
Almost none of this code should not be in WSM. Please look at the structure of Terra Policy Service (TPS).

Please read [AMALGAM_README](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/amalgam/AMALGAM_README.md) and follow that for how to structure LZ integration with WSM.

File: scripts/clean_landingzone.sql (hunk 4 above)
Comment (zykonda) [reply scripts/clean_landingzone.sql:2]: it is convenience script, similar to workspace and stairway table clean up scripts.

**Comment by zykonda** [comment]
The work has been restructured per suggestion.
A new repo has been created for Landing Zone component. 
Work for [TOAZ-184] made this PR irrelevant, so closing.

**Pull request closed without merging**